src/DynamicNLPModels.jl

```diff
@@ -333,17 +339,72 @@ function DenseLQDynamicModel(
     ul::V = (similar(s0, size(R, 1)) .= -Inf),
     uu::V = (similar(s0, size(R, 1)) .=  Inf),
     gl::V = (similar(s0, size(E, 1)) .= -Inf),
-    gu::V = (similar(s0, size(F, 1)) .= Inf)
+    gu::V = (similar(s0, size(F, 1)) .= Inf),
+    truncated = false
 ) where {T, V <: AbstractVector{T}, M <: AbstractMatrix{T}, MK <: Union{Nothing, AbstractMatrix{T}}}
 
     dnlp = LQDynamicData(
         s0, A, B, Q, R, N;
         Qf = Qf, S = S, E = E, F = F, K = K,
         sl = sl, su = su, ul = ul, uu = uu, gl = gl, gu = gu)
 
-    DenseLQDynamicModel(dnlp)
+    DenseLQDynamicModel(dnlp; truncated = truncated)
+end
+
+"""
+    LQJacobianOperator{T, V, M}
+
+Struct for storing the truncated Jacobian matrix. All data for the Jacobian can be stored
+in the first `nu` columns of `J`. This struct contains the needed data and storage arrays for
+calculating `Jx`, `J^T x`, and `J^T Sigma J`. `Jx` and `J^T x` are performed through extensions
+to `LinearAlgebra.mul!()`.
+
+---
+Attributes
+ - `Jac`: Matrix of first `nu` columns of the Jacobian
+ - `N`  : number of time steps
+ - `nu` : number of inputs
+ - `nc` : number of algebraic constraints of the form gl <= Es + Fu <= gu
+ - `nsc`: number of bounded state variables
+ - `nuc`: number of bounded input variables (if `K` is defined)
+ - `scaled_Jac`: placeholder to avoid allocaiton when calculating `Sigma J`
+ - `J1Bx` : vector for storing multiplications when doing `Jx`
+ - `J2Bx` : vector for storing multiplications when doing `Jx`
+ - `J3Bx` : vector for storing multiplications when doing `Jx`
+ - `J1BTx`: vector for storing multiplications when doing `J^T x`
+ - `J2BTx`: vector for storing multiplications when doing `J^T x`
+ - `J3BTx`: vector for storing multiplications when doing `J^T x`
+ - `J1B`  : matrix for storing multiplicaitons when doing `J^T Sigma J`
+ - `J2B`  : matrix for storing multiplicaitons when doing `J^T Sigma J`
+ - `J3B`  : matrix for storing multiplicaitons when doing `J^T Sigma J`
+"""
+mutable struct LQJacobianOperator{T, V, M} <: LinearOperators.AbstractLinearOperator{T}
```

Comment:
This structure should not be mutable (as a general rule of thumb, we should avoid using mutable struct wherever possible)

Reply:
@frapac I was concerned about resetting the storage vectors/matrices and whether that would throw an error if this was a `struct` rather than a `mutable struct`. Is changing `J1Bx` inside `mul!` permitted for an immutable `struct`

Reply:
Good question. But this is not a concerned for us. Here, `J1Bx` is a pointer, and we don't change the reference once the memory has been allocated. We update only the values in the memory chunk specified by the pointer, not the pointer itself. So in short, `J1Bx` is a mutable structure, but `LQJacobianOperator` does not have to be mutable.